protege-editor-owl/src/main/java/org/protege/editor/owl/model/entity/RandomProlong.java

```diff
@@ -1,53 +1,39 @@
 package org.protege.editor.owl.model.entity;
 
 import org.semanticweb.owlapi.model.OWLEntity;
-
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import uk.org.russet.identitas.*;
 
-public class RandomProlong extends AbstractIDGenerator implements AutoIDGenerator, Revertable {
+/**
```

Comment:
I suggest you remove `extends AbstractIDGenerator`. See my overall comment.

Reply:
Thank you @csnyulas and @matthewhorridge for your replies. We cannot remove the extend  AbstractIDGenerator class, because we do need the getPrefix() and getSuffix() methods.